flow/project.py

```diff
@@ -245,6 +245,122 @@ def make_bundles(operations, size=None):
+            raise ValueError("JobOperation cmd must be a callable or string.")
+        self._cmd = cmd
+
+        if directives is None:
+            directives = dict()  # default argument
+        else:
+            directives = dict(directives)  # explicit copy
+
+        # Keys which were explicitly set by the user, but are not evaluated by the
+        # template engine are cause for concern and might hint at a bug in the template
+        # script or ill-defined directives. We are therefore keeping track of all
+        # keys set by the user and check whether they have been evaluated by the template
+        # script engine later.
+        keys_set_by_user = set(directives)
+
+        # We use a special dictionary that allows us to track all keys that have been
+        # evaluated by the template engine and compare them to those explicitly set
+        # by the user. See also comment above.
+        self.directives = TrackGetItemDict(
+            {key: value for key, value in directives.items()})
+        self.directives._keys_set_by_user = keys_set_by_user
+
+    def __str__(self):
+        return "{}({})".format(self.name, self.job)
+
+    def __repr__(self):
+        return "{type}(name='{name}', job='{job}', cmd={cmd}, directives={directives})".format(
+                   type=type(self).__name__,
+                   name=self.name,
+                   job=str(self.job),
+                   cmd=repr(self.cmd),
+                   directives=self.directives)
+
+    def __hash__(self):
+        return int(sha1(self.id.encode('utf-8')).hexdigest(), 16)
+
+    def __eq__(self, other):
+        return self.id == other.id
+
+    @deprecated(deprecated_in="0.9", removed_in="0.11", current_version=__version__)
+    def get_id(self):
+        return self._id
+
+    @property
+    def id(self):
+        return self._id
+
+    @property
+    def cmd(self):
+        if callable(self._cmd):
+            # We allow cmd to be 'lazy' or an unevaluated function because
+            # in cases where a user uses the Python API without specifying
+            # a project entrypoint, running many operations is still valid.
+            # If we need to fork this will fail to generate a command and
+            # error, but not until then. If we don't fork then nothing errors,
+            # and the user gets the expected result.
+            return self._cmd()
+        else:
+            return self._cmd
+
+    def set_status(self, value):
+        "Store the operation's status."
+        self.job._project.document.setdefault('_status', dict())[self.id] = int(value)
+
+    def get_status(self):
+        "Retrieve the operation's last known status."
+        try:
+            return JobStatus(self.job._project.document['_status'][self.id])
+        except KeyError:
+            return JobStatus.unknown
+
+
+@deprecated(
+    deprecated_in="0.11", removed_in="0.13", current_version=__version__)
 class JobOperation(object):
```

Comment:
I would just make `JobOperation` inherit from `_JobOperation` then it should work without the need to keep all the code in two places.

Reply:
@b-butler I tried that but this code gives an error that super method requires a class not a function.
`super(JobOperation, self).__init__(*args)`

Reply:
It worked locally for me. Don't use super just do
````python
@deprecated(...)
class JobOperation(_JobOperation):
    """Docstring here."""
    pass
````